### PR TITLE
feat: Form.Item support layout 3

### PR DESCRIPTION
**Pull request by crazyair**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     x      |
| 🇨🇳 Chinese |     x      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: components/form/style/index.ts [hunk 1]
```diff
@@ -510,37 +510,44 @@ const genVerticalStyle: GenerateStyle<FormToken> = (token) => {
         [`${formItemCls}-control`]: {
           width: '100%',
         },
+        [`${formItemCls}-label,
+        .${rootPrefixCls}-col-24${formItemCls}-label,
```
Comment (MadCcc) [inline components/form/style/index.ts:514]:
```suggestion
        ${antCls}-col-24${formItemCls}-label,
```
下同

File: components/form/style/index.ts [hunk 2]
```diff
@@ -385,11 +385,11 @@ const genFormItemStyle: GenerateStyle<FormToken> = (token) => {
   };
 };
 
-const genHorizontalStyle: GenerateStyle<FormToken> = (token) => {
-  const { componentCls, formItemCls } = token;
+const genHorizontalStyle = (token: FormToken, className: string) => {
```
Comment (MadCcc) [inline components/form/style/index.ts:388]:
```suggestion
const genHorizontalStyle = (token: FormToken, className: string): CSSObject => {
```

File: components/form/style/index.ts (hunk 1 above)
Comment (crazyair) [reply components/form/style/index.ts]: done

File: components/form/style/index.ts (hunk 2 above)
Comment (crazyair) [reply components/form/style/index.ts]: done

**Review by MadCcc — APPROVED** [review]

**Pull request merged**